### PR TITLE
[Unittest/IIO] add negative cases - @open sesame 05/06 15:26

### PR DESCRIPTION
**Pull request by helloahn**
testcase for the logic with:
1. invalid frequency
2. invalid base dir path

Signed-off-by: HyoungJoo Ahn <hello.ahn@samsung.com>

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: tests/nnstreamer_source/unittest_src_iio.cc [hunk 1]
```diff
@@ -1564,6 +1564,85 @@ TEST (test_tensor_src_iio, data_verify_freq_generic_type)
...
+  dev0->log_file = g_build_filename (dev0->base_dir, "temp.log", NULL);
+  parse_launch =
+      g_strdup_printf
+      ("%s iio-base-dir=%s device-number=%d name=my-src-iio ! fakesink",
+      ELEMENT_NAME, dev0->iio_base_dir_sim, 0);
+  src_iio_pipeline = gst_parse_launch (parse_launch, NULL);
+  g_free (parse_launch);
+
+  /** get and verify the caps */
+  src_iio = gst_bin_get_by_name (GST_BIN (src_iio_pipeline), "my-src-iio");
```
Comment (wooksong) [inline tests/nnstreamer_source/unittest_src_iio.cc:1596]: parse_launch might be null when gst_parse_lunch fails. https://gstreamer.freedesktop.org/documentation/gstreamer/gstparse.html?gi-language=c#gst_parse_launch


Note that "you might get a return value that is not NULL even though the error is set. In this case there was a recoverable parsing error and you can try to play the pipeline."

Comment (helloahn) [reply tests/nnstreamer_source/unittest_src_iio.cc:1597]: `nullptr` checker has been added before calling `g_free`

File: tests/nnstreamer_source/unittest_src_iio.cc [hunk 2]
```diff
@@ -1564,6 +1564,86 @@ TEST (test_tensor_src_iio, data_verify_freq_generic_type)
...
+  /** setup properties */
+  g_object_set (src_iio, "iio-base-dir", "/not/existed/path", NULL);
+  g_object_set (src_iio, "dev-dir", dev0->dev_dir, NULL);
+  g_object_set (src_iio, "device", DEVICE_NAME, NULL);
+
+  status = gst_element_set_state (src_iio, GST_STATE_PAUSED);
+  EXPECT_NE (status, GST_STATE_CHANGE_NO_PREROLL);
+
+  /** teardown */
+  gst_harness_teardown (hrnss);
```
Comment (gichan-jang) [inline tests/nnstreamer_source/unittest_src_iio.cc:1644]: Need `gst_object_unref (src_iio);`?

Comment (myungjoo) [reply tests/nnstreamer_source/unittest_src_iio.cc:1644]: gst_harness_teardown appears to do it.
We may check if src_iio is available or not after LINE1644 in the code to make sure that it's freed. (don't just add unref there unless you sure that it's not freed at that point)

File: tests/nnstreamer_source/unittest_src_iio.cc (hunk 1 above)
Comment (myungjoo) [reply tests/nnstreamer_source/unittest_src_iio.cc:1597]: You need to check ```src_iio_pipeline```, not ```parse_launch``` at LINE 1593.

Comment (helloahn) [reply tests/nnstreamer_source/unittest_src_iio.cc:1694]: @myungjoo it's fixed

File: tests/nnstreamer_source/unittest_src_iio.cc (hunk 2 above)
Comment (helloahn) [reply tests/nnstreamer_source/unittest_src_iio.cc:1741]: I executed Valgrind to check mem leak. Unfortunately, there are some points which should be fixed. I'll make another PR that fixes the problem include this one.

**Pull request merged**